include/libmpu/mpu6050.hpp

```diff
@@ -1,5 +1,201 @@
+      : m_i2c(&i2c), m_address(device_address)
+  {
+  }
+
+  hal::result<accelerometer::read_t> driver_read() override
+  {
+    accelerometer::read_t acceleration;
+    constexpr uint16_t bytes_per_axis = 2;
+    constexpr uint8_t number_of_axis  = 3;
+
+    std::array<hal::byte, bytes_per_axis * number_of_axis> xyz_acceleration;
+    HAL_CHECK(hal::write_then_read(*m_i2c,
+                                   m_address,
+                                   std::array{ xyz_register },
+                                   xyz_acceleration,
+                                   hal::never_timeout()));
+    // First X-axis Byte (MSB first)
+    // =========================================================================
+    // Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+    //  XD15 | XD14  |  XD13 |  XD12 |  XD11 |  XD10 |  XD9  |  XD8
+    //
+    // Final X-axis Byte (LSB)
+    // =========================================================================
+    // Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+    //   XD7 |   XD6 |   XD5 |   XD4 |   XD3 |   XD2 |   XD1 |   XD0
+    //
+
+    // We simply shift and OR the bytes together to get them into a signed int
+    // 16 value.
+    int16_t x =
+        static_cast<int16_t>(xyz_acceleration[0] << 8 | xyz_acceleration[1]);
+    int16_t y =
+        static_cast<int16_t>(xyz_acceleration[2] << 8 | xyz_acceleration[3]);
+    int16_t z =
+        static_cast<int16_t>(xyz_acceleration[4] << 8 | xyz_acceleration[5]);
+
+    // Convert the 16 bit value into a floating point value m/S^2
+    constexpr float kMax =
+        static_cast<float>(std::numeric_limits<int16_t>::max());
+    constexpr float kMin =
+        static_cast<float>(std::numeric_limits<int16_t>::min());
+
+    auto int16_bounds  = std::make_pair(kMax, kMin);
+    auto posneg_bounds = std::make_pair(-m_gscale, m_gscale);
+
+    acceleration.x = hal::map(x, int16_bounds, posneg_bounds);
+    acceleration.y = hal::map(y, int16_bounds, posneg_bounds);
+    acceleration.z = hal::map(z, int16_bounds, posneg_bounds);
+
+    return acceleration;
+  }
+
+  hal::status active_mode(bool is_active = true)
+  {
+    constexpr auto sleep_mask = hal::bit::mask::from<6>();
+
+    auto control =
+        HAL_CHECK(hal::write_then_read<1>(*m_i2c,
+                                          m_address,
+                                          std::array{ initalizing_register },
+                                          hal::never_timeout()));
+
+    hal::bit::modify(control[0]).insert<sleep_mask>(!is_active);
+
+    HAL_CHECK(hal::write(*m_i2c,
+                         m_address,
+                         std::array{ initalizing_register, control[0] },
+                         hal::never_timeout()));
+
+    return hal::success();
+  }
+
+  hal::status is_valid_device()
+  {
+    // Verify that the device is the correct device
+    hal::byte expected_device_id = 0x68;
```

Comment:
```suggestion
    static constexpr hal::byte expected_device_id = 0x68;
```
Make this `static constexpr` so its known at compile time and is const correct.